c10/cuda/CMakeLists.txt

```diff
@@ -47,36 +51,42 @@ set(C10_CUDA_HEADERS
     impl/CUDATest.h
 )
 set(CUDA_LINK_LIBRARIES_KEYWORD PRIVATE)
-torch_cuda_based_add_library(c10_cuda ${C10_CUDA_SRCS} ${C10_CUDA_HEADERS})
-set(CUDA_LINK_LIBRARIES_KEYWORD)
-# If building shared library, set dllimport/dllexport proper.
-target_compile_options(c10_cuda PRIVATE "-DC10_CUDA_BUILD_MAIN_LIB")
-# Enable hidden visibility if compiler supports it.
-if(${COMPILER_SUPPORTS_HIDDEN_VISIBILITY})
-  target_compile_options(c10_cuda PRIVATE "-fvisibility=hidden")
-endif()
 
-# ---[ Dependency of c10_cuda
-target_link_libraries(c10_cuda PUBLIC c10 torch::cudart)
+if(NOT BUILD_LIBTORCHLESS)
+  torch_cuda_based_add_library(c10_cuda ${C10_CUDA_SRCS} ${C10_CUDA_HEADERS})
+  set(CUDA_LINK_LIBRARIES_KEYWORD)
+  # If building shared library, set dllimport/dllexport proper.
+  target_compile_options(c10_cuda PRIVATE "-DC10_CUDA_BUILD_MAIN_LIB")
+  # Enable hidden visibility if compiler supports it.
+  if(${COMPILER_SUPPORTS_HIDDEN_VISIBILITY})
+    target_compile_options(c10_cuda PRIVATE "-fvisibility=hidden")
+  endif()
 
-if(NOT WIN32)
-target_link_libraries(c10_cuda PRIVATE dl)
-target_compile_options(c10_cuda PRIVATE "-DPYTORCH_C10_DRIVER_API_SUPPORTED")
-endif()
+  # ---[ Dependency of c10_cuda
+  target_link_libraries(c10_cuda PUBLIC ${C10_LIB} torch::cudart)
 
-target_include_directories(
-    c10_cuda PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../..>
-    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
-    $<INSTALL_INTERFACE:include>)
+  if(NOT WIN32)
+  target_link_libraries(c10_cuda PRIVATE dl)
+  target_compile_options(c10_cuda PRIVATE "-DPYTORCH_C10_DRIVER_API_SUPPORTED")
+  endif()
 
-add_subdirectory(test)
+  target_include_directories(
+      c10_cuda PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../..>
+      $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>)
+  set(C10_CUDA_LIB c10_cuda)
```

Comment:
Why are we not doing this at the beginning of the file ? So that everywhere else its 
```
${C10_CUDA_LIB}
```

Reply:
https://github.com/pytorch/pytorch/pull/125249#pullrequestreview-2043971589 describes the purpose of these aliases as well.